proposals/i18n-system-languages.md

```diff
@@ -0,0 +1,107 @@
+# Proposal: i18n.getPreferredSystemLanguages() and i18n.getSystemUILanguage()
+
+**Summary**
+
+Allows getting the language(s)/locale(s) of your operating system as a [BCP47 language tag](https://www.rfc-editor.org/bcp/bcp47.html)
+
+**Document Metadata**
+
+**Author:** [carlosjeurissen](https://github.com/carlosjeurissen)
+
+**Sponsoring Browser:** Safari
+
+**Created:** 2024-03-18
+
+**Related Issues:**
+https://github.com/w3c/webextensions/issues/252
+https://bugzilla.mozilla.org/show_bug.cgi?id=1888486
+
+## Motivation
+
+### Objective
+
+Allow extensions to display UI in the user's system locale(s), even when this is distinct from the browser's UI locale (which is restricted to locales supported by the browser).
+
+#### Use Cases
+
+Extension authors might want to obtain the locale used by the user agent's host environment in order to format values (such as numbers, dates, and so forth) or do other locale-affected operations (such as list sorting) according to expectations of the user. For example, formatting the date `2024-06-15` varies between `en-US` (6/15/2024) and `en-GB` (15/6/2024), whereas `i18n.getUILanguage` might return only `en` (English).
+
+Extension authors might want to obtain the locale used by the user agent's host environment to better match the runtime environment's localization. For example, the operating system might be running in `ff-NG` (Fulani as used in Nigeria), but the user agent might not be localized for this locale and return `en` (English) for `i18n.getUILanguage`.
+
+Extension authors might want to obtain the locale used by the user agent's host environment in order to get the locales in it's purest form. For example, the operating system might be running in `es-AR` and a browser's closest equivalent is `es-419`. Having access to `es-AR` allows extension authors to provide better translations for all users.
+
+### Known Consumers
+
+Language-related extensions and extensions wanting to match the locale of the operating system more closely independent of the locale of the browser.
+
+## Specification
+
+### Schema
+
+`i18n.getPreferredSystemLanguages()` would asynchronously return a language priority list (ordered list of [BCP47 language tag](https://www.rfc-editor.org/bcp/bcp47.html). This is similar to `i18n.getAcceptLanguages()`, with the difference that the priority list is configured by the user in the OS and the method would not add less specific language tags in place.
+
+```json
+{
+  "name": "getPreferredSystemLanguages",
+  "type": "function",
+  "description": "Gets the preferred languages of the operating system. This is different from the languages set in the browser; to get those, use $(ref:i18n.getAcceptLanguages).",
```

Comment:
The introduction of this md document emphasizes the distinction between language and locale, but here the description of `getPreferredSystemLanguages` refers to language only. The description below mentions locale, based on the feedback from https://github.com/w3c/webextensions/pull/569#discussion_r1784857029

Unless you really intended to limit the meaning to "language (not locale)", let's try to be consistent.

Reply:
Updated the PR. Now states locale.